src/mmap.cc

```diff
@@ -31,21 +34,71 @@ static void DontFree(char* data, void* hint) {
 NAN_METHOD(Alloc) {
   Nan::HandleScope scope;
 
-  int len = info[0]->Uint32Value();
+  std::size_t len;
+
+  if (node::Buffer::HasInstance(info[0])) {
+    len = *reinterpret_cast<std::size_t *>(node::Buffer::Data(info[0]));
+  } else {
+    len = info[0]->Uint32Value();
```

Comment:
I think we could actually use `IntegerValue` to get 52 bits of precision instead of 32.

Reply:
@indutny Yes, good point.

\- Jonathan